8.0 new product attribute group

gdgellatly opened the pull request:
Add new module product_attribute_group to support the grouping of attributes for reuse within product templates.

pedrobaeza commented on product_attribute_group/README.rst:
```diff
@@ -0,0 +1,132 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Module name
```
Put the actual module name

pedrobaeza commented on product_attribute_group/README.rst:
```diff
@@ -0,0 +1,132 @@
+
+* do nothing, no changes are made to existing data and fields and no new dependencies are introduced.
+
+Uninstallation
```
This can be part of the same installation section

gdgellatly committed:
product_attribute_group

Add new module product_attribute_group to support the grouping of attributes for reuse within product templates.
Fix wrong copyright file
Fix flake8 failures
Fix error in demo data
Fix Readme as per Pedro
Little bit of cleanup and additional demo data
Add tests

gdgellatly commented:
@gurneyalex can you help me. I don't know what to do here.  Travis keeps failing, but here are my test results with product_sequence installed on this branch on a test database.  I have no idea why it doesn't call the product_sequence overrides on travis to create the codes. (You can see by the long runtime I spent a lot of time in debug ensuring that they are called)

2015-06-25 04:35:57,779 20656 INFO seq_test5 openerp.modules.module: openerp.addons.product_attribute_group.tests.test_attribute_group running tests.
2015-06-25 04:35:57,780 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test03a_action_update_variants_addition (openerp.addons.product_attribute_group.tests.test_attribute_group.EnhancedAttributeGroupTestCase)
2015-06-25 04:42:57,516 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test03b_action_update_variants_addition_overlap (openerp.addons.product_attribute_group.tests.test_attribute_group.EnhancedAttributeGroupTestCase)
2015-06-25 04:48:23,114 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test03c_action_update_variants_removal (openerp.addons.product_attribute_group.tests.test_attribute_group.EnhancedAttributeGroupTestCase)
2015-06-25 04:48:36,390 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test03d_action_update_variants_removal_overlap (openerp.addons.product_attribute_group.tests.test_attribute_group.EnhancedAttributeGroupTestCase)
2015-06-25 04:48:39,056 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test07a_onchange_value_ids_normal (openerp.addons.product_attribute_group.tests.test_attribute_group.EnhancedAttributeGroupTestCase)
2015-06-25 04:48:41,491 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test07b_onchange_value_ids_abnormal (openerp.addons.product_attribute_group.tests.test_attribute_group.EnhancedAttributeGroupTestCase)
2015-06-25 04:48:43,900 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test08_onchange_attribute_group_ids (openerp.addons.product_attribute_group.tests.test_attribute_group.EnhancedAttributeGroupTestCase)
2015-06-25 04:48:46,296 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test01_copy_attribute_group (openerp.addons.product_attribute_group.tests.test_attribute_group.SimpleAttributeGroupCase)
2015-06-25 04:48:46,343 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test02a_unlink_attribute_group_prohibited (openerp.addons.product_attribute_group.tests.test_attribute_group.SimpleAttributeGroupCase)
2015-06-25 04:48:46,661 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test02b_unlink_attribute_group_normal (openerp.addons.product_attribute_group.tests.test_attribute_group.SimpleAttributeGroupCase)
2015-06-25 04:48:46,693 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test04_onchange_attribute_value_ids (openerp.addons.product_attribute_group.tests.test_attribute_group.SimpleAttributeGroupCase)
2015-06-25 04:48:47,086 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test05a_onchange_attribute_id_prohibited (openerp.addons.product_attribute_group.tests.test_attribute_group.SimpleAttributeGroupCase)
2015-06-25 04:48:47,446 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test05b_onchange_attribute_id (openerp.addons.product_attribute_group.tests.test_attribute_group.SimpleAttributeGroupCase)
2015-06-25 04:48:47,468 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: test06_constraint_attribute_id (openerp.addons.product_attribute_group.tests.test_attribute_group.SimpleAttributeGroupCase)
2015-06-25 04:48:47,507 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: Ran 14 tests in 769.726s
2015-06-25 04:48:47,507 20656 INFO seq_test5 openerp.addons.product_attribute_group.tests.test_attribute_group: OK

gdgellatly commented:
@gurneyalex @bwrsandman Hi, I've retested this using the travis commands exactly on a local machine.  I think it is a real problem and we might need to change the test script.

Findings
If I run script with product_sequence included in --update - Fails
If I run script with product_sequence included in --update but update module to depend on product_sequence - Pass
If I run script without product_sequence and no dependency (but product_sequence is installed) - Pass

The upshot of this is that I think the test runner should loop over the directories and run --update (1 module) each time.  It seems that update does not load the modules to be updated before testing.  Even then it is still not ideal, as in theory an installed module could incorrectly cause a test to pass or fail.

gurneyalex commented:
@gdgellatly after a (long) while, rebuilding in travis is almost green : only lint errors remain. 

Are you still interested in this PR?

gdgellatly commented:
@gurneyalex Wow, sorry I missed this, actually I am still really interested but will retarget at 10

gdgellatly commented:
Closing as I'll never use it and a much simpler port to 10 in #254
